docs/design-proposals/multi-tenancy/multi-tenancy-schduler-agent-and-controllers.md

```diff
@@ -0,0 +1,69 @@
+# Scheduler, Node Agent and Controllers in Arktos Multi-Tenancy
+
+Xiaoning Ding, Qian Chen, Peng Du
+
+## Introduction
+
+This document is a part of the multi-tenancy design documents in Arktos. It describes most of the non-API-Server changes, including changes in scheduler, node agent and controllers.
+
+Please refer to other design documents in the "multi-tenancy" document directory if you want to learn about other parts in the multi-tenancy design.
+
+## Scheduler
+
+Scheduler and controllers work in a similar way. They all list/watch some API objects and then make update to API certain objects. This section discusses some common changes that are required in schedulers and controllers to make them tenancy-aware. And also a new tenant controller.
+
+### General Tenancy-aware changes
+
+Scheduler and controllers need to use right client-go APIs to list/watch all objects, or update an object in a certain world. Also world-matching is required in their logics.
+
+Take the scheduler as an example:
+
+* When list/watch objects like pods or pvs, it should list/watch all objects in system world and tenant worlds using credentials of system user. This is achieved by calling client-go API with TenantAll:
```

Comment:
system world and tenant worlds --> system space and tenant spaces

Reply:
Forgot to change here. Will change.

Reply:
Updated.